test/common_methods_invocations.py

```diff
@@ -980,6 +980,45 @@ def run_additional_tri_tests(self, device):
             1, 1, device=device, layout=torch.sparse_coo))
 
 
+def _compare_any_results(self, device):
```

Comment:
I don't think these tests should be here (I know the triu tests are here too, but I don't think those should be here either.

This test file is to have standard tests we want to run across test types, e.g. autograd / JIT, so they need to be self runnable.  Tests that invoke common_method_invocations can't automatically know that a device needs to be passed here.

I think a better way to do this is just put the device type in a loop in test_torch.py, e.g.:
https://github.com/pytorch/pytorch/blob/bee6c6761e9e933df5fdebc0481ec790d9011a41/test/test_torch.py#L1099-L1100

It has some downsides in that when we add devices we'll need to change this, but it's a lot nicer in that all the testing code for all devices is together.

Reply:
I tried that before (~ a month ago) I think, but later realized that our ci framework will not explicitly schedule `test_torch.py` onto gpu servers. So, the `cuda` test either hits an error or is skipped. 

Is there any way that I can force a test to be scheduled to gpu servers?

Reply:
After some thinking, I feel that how ci schedules tests is out of the scope of this PR, and I should not try to hack it by placing the test to inappropriate positions. 

Thanks for the suggestions. I will revise accordingly.

Reply:
@gchanan @mrshenli We do run all tests in `test_torch.py` by default on GPU machines, unless they are guarded by `@unittest.skipIf(not torch.cuda.is_available(), 'no CUDA')`

Reply:
I see, thanks!  @gchanan @yf225